Update Zoo.cs - Exception handling

CCianfloneDev opened the pull request:
Added documentation and Impl. exception handling. For Issue #12

CCianfloneDev committed:
Update Zoo.cs

Added documentation and Impl. exception handling.

CCianfloneDev committed:
Update Zoo.cs

DZ-FSDev commented on ZooKeepingSystem/Zoo.cs:
```diff
@@ -38,19 +50,38 @@ public string ZooName
         /// <param name="numberOfZooKeepers">Total number of zoo keepers employed.</param>
         /// <param name="animals">Animals.</param>
         /// <param name="zooName">Name of the zoo.</param>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when initialized with no name.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when number of cages or number of zoo keepers is less than zero.
+        /// </exception>
         public Zoo(string zooName, int numberOfCages, int numberOfZooKeepers, Animals animals)
         {
+            if (zooName == null)
+            {
+                throw new ArgumentNullException("Name", "Name cannot be null.");
+            }
+            if (numberOfCages < 0)
+            {
+                throw new ArgumentOutOfRangeException("NumberOfCages", "Argument cannot be less than zero.");
+            }
+            if (numberOfZooKeepers < 0)
+            {
+                throw new ArgumentOutOfRangeException("NumberOfZooKeepers", "Argument cannot be less than zero.");
+            }
+
             ZooName = zooName;
             NumberOfCages = numberOfCages;
             NumberOfZooKeepers = numberOfZooKeepers;
             AnimalsInZoo = animals;
         }
 
         /// <summary>
-        /// Initializes a zoo with a name, 0 cages, 1 zoo keeper, and no animals in the care.
+        /// Initializes a zoo with a name, 1 cage, 1 zoo keeper, and no animals in the care.
         /// </summary>
         /// <param name="zooName">The name of the zoo.</param>
-        public Zoo(string zooName) : this(zooName, 0, 1, null)
+        public Zoo(string zooName) : this(zooName, 1, 1, null)
```
- [x] Add exception documentation from invoking `this(zooName, 1, 1, null)`

DZ-FSDev commented on ZooKeepingSystem/Zoo.cs:
```diff
@@ -38,19 +50,38 @@ public string ZooName
+        /// </exception>
         public Zoo(string zooName, int numberOfCages, int numberOfZooKeepers, Animals animals)
         {
+            if (zooName == null)
+            {
+                throw new ArgumentNullException("Name", "Name cannot be null.");
+            }
+            if (numberOfCages < 0)
+            {
+                throw new ArgumentOutOfRangeException("NumberOfCages", "Argument cannot be less than zero.");
+            }
+            if (numberOfZooKeepers < 0)
+            {
+                throw new ArgumentOutOfRangeException("NumberOfZooKeepers", "Argument cannot be less than zero.");
+            }
```
- [x] The logic for state validation should be in the property set accessors.

DZ-FSDev commented on ZooKeepingSystem/Zoo.cs:
```diff
@@ -38,19 +50,38 @@ public string ZooName
         {
+            if (zooName == null)
+            {
+                throw new ArgumentNullException("Name", "Name cannot be null.");
```
- [x] Change param name from `Name` to `zooName`
- [x] zooName also should not be blank i.e. `""`

DZ-FSDev commented on ZooKeepingSystem/Zoo.cs:
```diff
@@ -38,19 +50,38 @@ public string ZooName
+            }
+            if (numberOfCages < 0)
+            {
+                throw new ArgumentOutOfRangeException("NumberOfCages", "Argument cannot be less than zero.");
```
- [x] Change param name to camel case

DZ-FSDev commented on ZooKeepingSystem/Zoo.cs:
```diff
@@ -38,19 +50,38 @@ public string ZooName
+            }
+            if (numberOfZooKeepers < 0)
+            {
+                throw new ArgumentOutOfRangeException("NumberOfZooKeepers", "Argument cannot be less than zero.");
```
- [x] Change NumberOfZooKeepers to camel case

DZ-FSDev commented on ZooKeepingSystem/Zoo.cs:
```diff
@@ -7,28 +7,40 @@ namespace ZooKeepingSystem
         }
 
+        /// <summary>
+        /// Gets and sets the name of the zoo.
```
- [x] Doesn't have a `public set` so should be documented as a `get` only

DZ-FSDev reviewed (changes requested):
**CR for #20**

CCianfloneDev committed:
Update Zoo.cs

Added exception documentation for generic constructor.

CCianfloneDev committed:
Update Zoo.cs

Fixed recursive properties. Correct documentation, moved exceptions from constructor to properties.

DZ-FSDev commented on ZooKeepingSystem/Zoo.cs:
```diff
@@ -7,22 +7,46 @@ namespace ZooKeepingSystem
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException("numberOfCages", "Argument cannot be less than zero.");
```
- [x] The argument referenced in property setters is usually `value`

DZ-FSDev commented on ZooKeepingSystem/Zoo.cs:
```diff
@@ -7,22 +7,46 @@ namespace ZooKeepingSystem
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException("numberOfZooKeepers", "Argument cannot be less than zero.");
```
- [x] The argument referenced in property setters is usually `value`

DZ-FSDev commented on ZooKeepingSystem/Zoo.cs:
```diff
@@ -35,12 +59,22 @@ public Animals AnimalsInZoo
+            {
+                if (String.IsNullOrEmpty(value))
+                {
+                    throw new ArgumentNullException("zooName", "Name cannot be null.");
```
- [x] The argument referenced in property setters is usually `value`

DZ-FSDev commented on ZooKeepingSystem/Zoo.cs:
```diff
@@ -58,29 +92,22 @@ public string ZooName
 
         /// <summary>
-        /// Initializes a zoo with a name, 1 cage, 1 zoo keeper, and no animals in the care.
+        /// Initializes a zoo with a name, 1 cages, 1 zoo keeper, and no animals in the care.
```
- [x] 1 cage.

DZ-FSDev commented on ZooKeepingSystem/Zoo.cs:
```diff
@@ -58,29 +92,22 @@ public string ZooName
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when initialized with no name.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when number of cages or number of zoo keepers is less than zero.
+        /// </exception>
```
- [x] This exception shouldn't be thrown by this constructor.

DZ-FSDev reviewed (changes requested):
**CR for #20**

CCianfloneDev committed:
Update Zoo.cs

Corrected param. name of exceptions

CCianfloneDev committed:
Update Zoo.cs

Corrected documentation

DZ-FSDev reviewed (approved):
Looks good!

CCianfloneDev commented:
Exception handl. successfully impl.
All menus are functioning as intended.

CCianfloneDev commented:
Accidentally closed PR before merging, I reopened then merged, lol.
